docs/cookbook/images-gridfs.rst

```diff
@@ -0,0 +1,630 @@
+
+            $queryBuilder = $this->getRepository()->createQueryBuilder();
+
+            $queryBuilder
+                ->remove()
+                ->multiple()
+                ->field('metadata.filter')
+                ->in($filters);
+
+            if (! empty($paths)) {
+                $queryBuilder
+                    ->field('filename')
+                    ->in($paths);
+            }
+
+            $queryBuilder->getQuery()->execute();
+        }
+
+        /**
+         * @return DocumentRepository
+         */
+        private function getRepository(): DocumentRepository
+        {
+            return $this->documentManager->getRepository($this->class);
+        }
+    }
+
+.. code-block:: php
+
+    namespace AppBundle\Imagine\Cache\Resolver;
+
+    class GridFSException extends \RuntimeException
+    {
+    }
+
+.. code-block:: yml
+
+    app.imagine_resolver.grid_fs:
+      class: AppBundle\Imagine\Cache\Resolver\GridFSResolver
+      arguments:
+        - "@doctrine.odm.mongodb.document_manager"
+        - "AppBundle\\Document\\Product\\Image\\Cache"
+        - "@router"
+      tags:
+        - { name: "liip_imagine.cache.resolver", resolver: app.imagine_resolver.grid_fs }
+
+Now we're going to add a new controller (or action) resolving a requested cached product image.
+
+.. tip::
+
+    This implementation uses the `Symfony Doctrine param converter <https://symfony.com/doc/current/bundles/SensioFrameworkExtraBundle/annotations/converters.html#doctrine-converter>`_.
+
+.. code-block:: php
+
+    namespace AppBundle\Controller;
+
+    use AppBundle\Document\Product\Image\Cache;
+    use FOS\RestBundle\Controller\Annotations\Get;
+    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+
+    class ImagineController extends Controller
+    {
+        /**
+         * @Get("/product/media/cache/{id}", name="product_cache_image")
+         *
+         * @param Cache $cache
+         * @param Request $request
+         *
+         * @return Response
+         */
+        public function imageAction(Cache $cache, Request $request)
+        {
+            $response = new Response();
+            $response->setEtag($cache->getMd5());
+
+            if ($response->isNotModified($request)) {
+                return $response;
+            }
+
+            $disposition = $response->headers->makeDisposition(ResponseHeaderBag::DISPOSITION_INLINE, $cache->getId());
+            $response->headers->set('Content-Disposition', $disposition);
+            $response->headers->set('Content-Type', $cache->getContentType());
+
+            $response->setContent($cache->getFile()->getBytes());
+
+            return $response;
+        }
+    }
+
+Last but not least ``liip_imagine`` is reconfigured in you ``app/config/config.yml``.
+
+.. code-block:: yml
+
+    liip_imagine:
```

Comment:
Move this code block below the definition for `app.imagine_resolver.grid_fs` and add a 'enabling it ...' text?